android/envoy/src/main/kotlin/org/greatfire/envoy/EnvoyOkClient.kt

```diff
@@ -0,0 +1,129 @@
+
+package org.greatfire.envoy
+
+import android.net.Uri
+import android.util.Log
+import health.flo.network.ohttp.client.IsOhttpEnabledProvider
+import health.flo.network.ohttp.client.OhttpConfig
+import health.flo.network.ohttp.client.setupOhttp
+import java.io.File
+import java.net.Proxy
+import java.util.Collections
+import java.util.concurrent.TimeUnit
+import javax.net.ssl.SSLContext
+import javax.net.ssl.TrustManager
+import okhttp3.*
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import org.conscrypt.Conscrypt
+
+
+class EnvoyOkClient {
+    companion object {
+        private val TAG = "EnvoyOkClient"
+
+        private fun getClientBuilder(state: EnvoyState, timeout: Long?): OkHttpClient.Builder {
+            val builder = OkHttpClient.Builder()
+
+            val tm = Conscrypt.getDefaultX509TrustManager()
+
+            val sslContext: SSLContext = SSLContext.getInstance("SSL").apply {
+                init(null, arrayOf<TrustManager>(tm), java.security.SecureRandom())
+            }
+
+            builder
+                .connectionSpecs(Collections.singletonList(
+                    ConnectionSpec.Builder(ConnectionSpec.MODERN_TLS)
+                        .tlsVersions(TlsVersion.TLS_1_3).build())
+                )
+                .sslSocketFactory(sslContext.getSocketFactory(), tm)
+
+            if (timeout != null) {
+                builder.callTimeout(timeout, TimeUnit.SECONDS)
+            }
+
+            if (state.concealedAuthUser != ""
+                && state.concealedAuthPrivateKey != null
+                && state.concealedAuthPublicKey != null)
+            {
+                // We have Cocealed Auth config, so add the Interceptor
+                val interceptor = HTTPConcealedAuthInterceptor(
```

Comment:
Now that I think about it, we probably don't want to add this Interceptor (which adds and Authorization header) to every single client we use when Envoy is configured to support HTTP Concealed Auth. I don't think it should hurt anything, but it seems kind of silly to send it to every site... I guess I'll think about fixing that up